### ISSUE TITLE
Update to Downshift v7

### ISSUE THREAD

**Issue by jpveooys** [Royal-Navy/design-system#3565]
Downshift v7 has various changes related to ARIA 1.2: https://github.com/downshift-js/downshift/blob/master/src/hooks/MIGRATION_V7.md

The changes for useSelect are not insignificant (the wrapper itself should act as the toggle button, from what I can see).

This is also needed to fix failures with axe 4.5.
